### PR TITLE
Adjusted schemas

### PR DESCRIPTION
**Pull request by cohnjesse**
## Description
Original ticket - https://github.com/department-of-veterans-affairs/va.gov-team/issues/4640

In order to get the form 686 work from the Rainbows team working on staging we need to fix the form's submitted data structure so that it both matches the shape of the JSON schema for the back end AND is valid for the VA form system.

## Testing done
Unit tests will be included in a separate branch

## Acceptance criteria
- [x] The shape of the form data on submission matches the shape of the JSON schema object for the back end
- [x] No schema invalidations errors get thrown when clicking the submit button

### CODE REVIEW

File: src/applications/disability-benefits/686/config/form.js [hunk 1]
```diff
@@ -69,6 +69,7 @@ const {
   domesticAddress,
   fullName,
   internationalAddressText,
+  genericLocation,
```
Comment (micahchiang) [inline src/applications/disability-benefits/686/config/form.js]: Should this be removed? I was under the assumption we were getting rid of it moving forward.

File: src/applications/disability-benefits/686/config/form.js [hunk 2]
```diff
@@ -269,41 +277,16 @@ function createLocationUISchemaForKey(
 ) {
   return {
     'ui:title': title,
-    countryDropdown: {
-      'ui:title': 'Country',
+    state: {
+      'ui:title': 'State (or country if outside the USA',
       'ui:required': isRequiredCallback,
```
Comment (micahchiang) [inline src/applications/disability-benefits/686/config/form.js]: Can this property be removed if we're always asking for city and state (or country) moving forward?

File: src/applications/disability-benefits/686/config/form.js [hunk 3]
```diff
@@ -269,41 +277,16 @@ function createLocationUISchemaForKey(
...
-    state: {
-      'ui:title': 'State',
-      'ui:required': (formData, index) =>
-        isUSAAddress(get(`${insertRealIndexInKey(key, index)}`, formData)),
-      'ui:options': {
-        hideIf: (formData, index) =>
-          !isUSAAddress(get(`${insertRealIndexInKey(key, index)}`, formData)),
-      },
+      'ui:title': 'City or county', 
+      'ui:required': isRequiredCallback,
```
Comment (micahchiang) [inline src/applications/disability-benefits/686/config/form.js]: Can this property be removed if we're always asking for city and state (or country) moving forward?

File: src/platform/forms-system/src/js/validation.js [hunk 4]
```diff
@@ -267,8 +267,14 @@ export function isValidForm(form, pageList) {
...
       }
 
+      console.log(schema);
+      console.log(formData);
+
       const result = v.validate(formData, schema);
 
+      console.log(result);
+      console.log();
+
```
Comment (micahchiang) [inline src/platform/forms-system/src/js/validation.js]: These console logs should be removed.

**Review by micahchiang — CHANGES_REQUESTED** [review]
Some questions about a couple of the properties, and also a comment to remove the console logs in `validation.js`.

File: src/applications/disability-benefits/686/config/form.js [hunk 5]
```diff
@@ -269,40 +274,13 @@ function createLocationUISchemaForKey(
 ) {
   return {
     'ui:title': title,
-    countryDropdown: {
-      'ui:title': 'Country',
+    state: {
+      'ui:title': 'State (or country if outside the USA',
```
Comment (micahchiang) [inline src/applications/disability-benefits/686/config/form.js]: I think this is missing an ending paren ')' after USA.

**Review by micahchiang — APPROVED** [review]
LGTM

**Review by cvalarida — APPROVED** [review]

**Pull request merged**